hipcheck/src/plugin/manager.rs

```diff
@@ -3,13 +3,106 @@
 use crate::{
 	hc_error,
 	plugin::{try_get_bin_for_entrypoint, HcPluginClient, Plugin, PluginContext},
+	policy::policy_file::PolicyPluginName,
 	Result,
 };
 use futures::future::join_all;
 use hipcheck_common::proto::plugin_service_client::PluginServiceClient;
+use log::{debug, error, info, trace, warn};
 use rand::Rng;
-use std::{ffi::OsString, ops::Range, path::Path, process::Command};
-use tokio::time::{sleep_until, Duration, Instant};
+use serde::Deserialize;
+use std::{collections::HashMap, env, ffi::OsString, ops::Range, path::Path, sync::LazyLock};
+use tokio::process::Command;
+use tokio::{
+	io::{AsyncBufReadExt, BufReader},
+	time::{sleep_until, Duration, Instant},
+};
+
+static PLUGIN_LOG_SETTINGS: LazyLock<PluginLogLevels> =
+	LazyLock::new(|| PluginLogLevels::new().expect("Failed to parse Plugin log levels"));
+
+/// structs for deserializing plugin logs
+#[derive(Debug, Deserialize)]
+struct LogEvent {
+	level: LogLevel,
+	fields: LogFields,
+	target: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct LogFields {
+	message: String,
+}
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+enum LogLevel {
+	Off,
+	Error,
+	Warn,
+	Info,
+	Debug,
+	Trace,
+}
+
+pub struct PluginLogLevels {
+	plugin_wide_level: String,
+	plugin_level_map: HashMap<String, String>,
+}
+impl PluginLogLevels {
+	/// Parses HC_LOG env var to fetch log level for each plugin as Struct
+	pub fn new() -> Result<Self> {
+		let mut plugin_wide_level = String::from("off");
+		let mut plugin_level_map: HashMap<String, String> = HashMap::new();
+
+		if let Ok(rust_log) = env::var("HC_LOG") {
+			let settings: Vec<&str> = rust_log.split(',').collect();
+			let mut plugin_wide_set = false;
+
+			for setting in settings {
+				let parts: Vec<&str> = setting.split('=').collect();
+
+				if parts.len() == 1 {
```

Comment:
Would you please comment this block and the `parts.len() == 2` block with an explanation of what these cases mean, with at least one example for each? I think that would help a lot, since a reader of this code isn't necessarily familiar with the syntax this code is trying to parse.

Reply:
I think a match construction here would also be clearer: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=501b8472e9c6899e95ad773bca5dd687

Reply:
Thanks for adding comments and refactoring to use `match`!